codegen/value.go

```diff
@@ -201,7 +202,14 @@ func (v *fsValue) Filesystem() (Filesystem, error) {
 }
 
 func (v *fsValue) Request() (solver.Request, error) {
-	def, err := v.fs.State.Marshal(context.Background(), llb.LinuxAmd64)
+	// TODO this should probably derive from DefaultPlatform(ctx) but we
+	// dont have ctx in scope here. Not sure of the implications of adding
+	// ctx to the API yet.
```

Comment:
Would it be simpler to add a `defaultPlatform` field to `fsValue` and populate it when we create one a `fsValue`?